File: backend/app/schemas/hospital.py
from pydantic import BaseModel
from typing import Optional
from enum import Enum
from datetime import datetime

class HospitalStatusEnum(str, Enum):
    ACTIVE = "ACTIVE"
    BUSY = "BUSY"
    FULL = "FULL"
    OFFLINE = "OFFLINE"

class HospitalCreate(BaseModel):
    hospitalId: str
    name: str
    phoneNumber: str
    email: str
    address: str
    latitude: float
    longitude: float
    totalBeds: int
    availableBeds: int
    totalICUBeds: int
    availableICUBeds: int
    totalEmergencyBeds: int
    availableEmergencyBeds: int
    status: HospitalStatusEnum

class HospitalResponse(BaseModel):
    id: str
    hospitalId: str
    name: str
    phoneNumber: str
    email: str
    address: str
    latitude: float
    longitude: float
    totalBeds: int
    availableBeds: int
    totalICUBeds: int
    availableICUBeds: int
    totalEmergencyBeds: int
    availableEmergencyBeds: int
    status: str
    createdAt: datetime
    updatedAt: datetime

    class Config:
        from_attributes = True

class CapacityUpdate(BaseModel):
    availableBeds: Optional[int] = None
    availableICUBeds: Optional[int] = None
    availableEmergencyBeds: Optional[int] = None

File: frontend/lib/features/hospital/domain/entities/hospital.dart
import 'hospital_status.dart';

class Hospital {
  final String id;
  final String hospitalId;
  final String name;
  final String phoneNumber;
  final String email;
  final String address;
  final double latitude;
  final double longitude;
  final int totalBeds;
  final int availableBeds;
  final int totalICUBeds;
  final int availableICUBeds;
  final int totalEmergencyBeds;
  final int availableEmergencyBeds;
  final HospitalStatus status;
  final DateTime createdAt;
  final DateTime updatedAt;

  const Hospital({
    required this.id,
    required this.hospitalId,
    required this.name,
    required this.phoneNumber,
    required this.email,
    required this.address,
    required this.latitude,
    required this.longitude,
    required this.totalBeds,
    required this.availableBeds,
    required this.totalICUBeds,
    required this.availableICUBeds,
    required this.totalEmergencyBeds,
    required this.availableEmergencyBeds,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
  });

  Hospital copyWith({
    String? id,
    String? hospitalId,
    String? name,
    String? phoneNumber,
    String? email,
    String? address,
    double? latitude,
    double? longitude,
    int? totalBeds,
    int? availableBeds,
    int? totalICUBeds,
    int? availableICUBeds,
    int? totalEmergencyBeds,
    int? availableEmergencyBeds,
    HospitalStatus? status,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return Hospital(
      id: id ?? this.id,
      hospitalId: hospitalId ?? this.hospitalId,
      name: name ?? this.name,
      phoneNumber: phoneNumber ?? this.phoneNumber,
      email: email ?? this.email,
      address: address ?? this.address,
      latitude: latitude ?? this.latitude,
      longitude: longitude ?? this.longitude,
      totalBeds: totalBeds ?? this.totalBeds,
      availableBeds: availableBeds ?? this.availableBeds,
      totalICUBeds: totalICUBeds ?? this.totalICUBeds,
      availableICUBeds: availableICUBeds ?? this.availableICUBeds,
      totalEmergencyBeds: totalEmergencyBeds ?? this.totalEmergencyBeds,
      availableEmergencyBeds: availableEmergencyBeds ?? this.availableEmergencyBeds,
      status: status ?? this.status,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}
